src/cascadia/TerminalApp/Toast.h

```diff
@@ -0,0 +1,38 @@
+/*++
+Copyright (c) Microsoft Corporation
+Licensed under the MIT license.
+
+Module Name:
+- Toast.h
+
+Module Description:
+- This is a helper class for wrapping a TeachingTip with a timer to
+  automatically dismiss it. Callers should add the TeachingTip wherever they'd
+  like in the UI tree, then wrap that TeachingTip in a toast like so:
+
+```
+std::unique_ptr<Toast> myToast = std::make_unique<Toast>(MyTeachingTip());
+```
+- Then, you can show the TeachingTip with
+
+```
+myToast->Open();
+```
+
+  which will open the tip and close it after a brief timeout.
+
+--*/
+
+#pragma once
+#include "pch.h"
+
+class Toast : public std::enable_shared_from_this<Toast>

```

Comment:
Curious, why in TermApp instead of TermControl?

Also, I bet this isn't accessible :/. We should probably figure out a way to expose this same information using UIA. If we do this in TermControl, that might be easier? Because we would just post this info in the `TermControlAutomationPeer`?

Reply:
Oh wait. I'm dumb. We're getting info on the window, not the TermControl. I think there might be no way to make this accessible, unfortunately :/. I'll add this to my list of things to come back to.

Reply:
I mean if this ain't accessible, it's because `TeachingTip` isn't. It's literally just a `TeachingTip` that auto-dismisses

Reply:
We can use an accessibility notification to state the window name.